app/src/main/java/com/example/awesomefish/ui/dialogs/HighScoreBottomSheet.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.awesomefish.ui.dialogs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.fragment.app.FragmentManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.awesomefish.R
+import com.example.awesomefish.di.DI
+import com.example.awesomefish.ui.lists.HighscoreListAdapter
+import com.example.awesomefish.utils.RoundedBottomSheetDialogFragment
+import kotlinx.android.synthetic.main.highscores.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+private const val HIGH_SCORE_TAG = "highScore"
+
+class HighScoreBottomSheet : RoundedBottomSheetDialogFragment() {
+
+    private val repository = DI.provideScoreRepository()
+    lateinit var emptyText: TextView
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.highscore_bottom_sheet, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        emptyText = view.findViewById(R.id.empty_text)
+        GlobalScope.launch {
+            withContext(Dispatchers.Main) {
+                val scores =
+                    repository.getAll().sortedBy { it.scoreValue }
```

Comment:
while we are here, lets' make this one liner

Reply:
alright